### DISCUSSION TITLE
v3 tests not discovered in Rider?

### DISCUSSION THREAD

**Discussion by johnnyggalt** [xunit/xunit#3042] [Question] [answered]
Hi,

I'm trying to port from v2 to v3 and my tests are not discovered in Rider using latest on NuGet (`0.4.0-pre.20`). Even if I create a new project using the template there are no tests discovered. Despite being pre-release, I cannot find anything to suggest v3 is incompatible with Rider, but maybe I'm expecting too much? I literally did:

```powershell
dotnet new install xunit.v3.templates
dotnet new xunit3
```

Opened the resulting project file in Rider, built the project, then checked Tests pane:

![image](https://github.com/user-attachments/assets/5b5eb709-3c2e-45ff-891e-bc9d5dfd3955)

Does anyone have v3 tests being discovered by Rider? I've tried both Metadata and Test Runner discovery modes:

![image](https://github.com/user-attachments/assets/a6ffc610-5547-4444-b519-a2dbfbaa1fd5)

Also tried the same thing with v2 just to be sure, and it works fine:

![image](https://github.com/user-attachments/assets/fb4b43e4-95f9-48d6-9717-2e9625e333ca)

Does anyone know what's going on?

**Comment by bradwilson** [comment]
Working fine here:

![image](https://github.com/user-attachments/assets/f27ee903-e670-4a2f-9e45-684dba07815c)

Test discovery is in Metadata mode (the default).

**Comment by johnnyggalt** [comment, marked as answer]
@bradwilson Thanks for responding (and for all the incredible work you put into xunit).

How interesting. I think this comes down to Rider version, and my lack of familiarity with it. I updated just the other day, so I assumed I had the latest. But I hadn't updated in a long while, and perhaps Rider doesn't automatically upgrade across major versions because my version was still older than yours:

![image](https://github.com/user-attachments/assets/2735fdad-c23a-4ac0-be69-e749c35fc8a2)

I enabled logging in Rider and could see this:

```
Warning: Exploration finished for <Tests,net8.0,xUnit> using XUnitTestExplorerFromArtifacts: Faulted (NuGet package 'Microsoft.NET.Test.Sdk' is not installed)
Warning: Exploration finished for <Tests,net8.0,xUnit>: Faulted ((+0 ~0 -0), interrupted: 0)
Warning: Exploration finished for <Tests,net8.0>: Faulted
Warning: Exploration finished for <Tests>: Faulted
```

Once I updated to _actual_ latest, test discovery works fine 😃
